### PR TITLE
Fix Review

### PR DESCRIPTION
**Pull request by sherlock-admin4**
Fix Review of 

Repo: `saffron-finance/lido-fiv`
Commit Hash: `bb4f953afa69612b8c38247767b441966fa1fe0e`

### CODE REVIEW

File: lido-fiv/contracts/LidoVault.sol [hunk 1]
```diff
@@ -47,15 +47,15 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
   uint256 public appliedProtocolFee;
 
   /// @notice ETH amount that tracks the total Saffron protocol fee from variable withdrawal staking earnings while the vault is still ongoing
-  uint256 public totalProtocolFee;
+  uint256 public ongoingProtocolFeeInShares;
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:50]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 2]
```diff
@@ -123,6 +123,9 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
   /// @notice Amount of ETH used to cover the returning of fixed user's initial principal
   uint256 public vaultEndedFixedDepositsFunds;
 
+  /// @notice Amount of earnings in ETH from Lido Staking after the vault has ended that was already withdrawn
+  uint256 public vaultEndedWithdrawnStakingEarnings;
+
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:128]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 3]
```diff
@@ -176,7 +179,7 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
   mapping(address => uint256) public variableToWithdrawnStakingEarningsInShares;
 
   /// @notice Mapping from variable side user addresses to their total payed protocol fee amount right after their Lido withdrawal request is submitted
-  mapping(address => uint256) public variableToWithdrawnProtocolFee;
+  mapping(address => uint256) public variableToWithdrawnProtocolFeeInShares;
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:182]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 4]
```diff
@@ -573,6 +572,19 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+    require(bearerBalance > 0, "NBT");
+    if (feeEarnings > 0) {
+      uint256 feeEarningsShare = calculateVariableFeeEarningsShare();
+      if (feeEarningsShare > 0) {
+        transferWithdrawnFunds(msg.sender, feeEarningsShare);
+        emit VariableFundsWithdrawn(feeEarningsShare, msg.sender, true, false);
+      }
+    }
+  }
+
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:587]: Change related to https://github.com/saffron-finance/lido-fiv/pull/30. 

Fixes issues: [#105](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/105)

File: lido-fiv/contracts/LidoVault.sol [hunk 5]
```diff
@@ -771,17 +783,28 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
-
-      if (totalEarnings > 0) {
-        (uint256 currentState, uint256 stakingEarningsShare) = calculateVariableWithdrawState(
-          totalEarnings,
-          variableToWithdrawnStakingEarningsInShares[msg.sender].mulDiv(vaultEndingETHBalance, vaultEndingStakesAmount)
-        );
-        stakingShareAmount = stakingEarningsShare;
-        variableToWithdrawnStakingEarningsInShares[msg.sender] = currentState.mulDiv(vaultEndingStakesAmount,vaultEndingETHBalance);
-        variableToWithdrawnStakingEarnings[msg.sender] = currentState;
+      if (vaultEndingETHBalance >= MIN_STETH_WITHDRAWAL_AMOUNT) {
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:799]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 6]
```diff
@@ -771,17 +783,28 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+        uint256 totalEarnings = vaultEndingETHBalance.mulDiv(
+          withdrawnStakingEarningsInStakes - ongoingProtocolFeeInShares,
+          vaultEndingStakesAmount
+        ) + vaultEndedStakingEarnings;
+        if (totalEarnings > 0) {
+          (uint256 currentState, uint256 stakingEarningsShare) = calculateVariableWithdrawState(
+            totalEarnings,
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:806]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 7]
```diff
@@ -771,17 +783,28 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+            (variableToWithdrawnStakingEarningsInShares[msg.sender] -
+              variableToWithdrawnProtocolFeeInShares[msg.sender]).mulDiv(vaultEndingETHBalance, vaultEndingStakesAmount)
+          );
+          stakingShareAmount = Math.min(
+            stakingEarningsShare,
+            vaultEndedStakingEarnings - vaultEndedWithdrawnStakingEarnings
+          );
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:813]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 8]
```diff
@@ -771,17 +783,28 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+          vaultEndedWithdrawnStakingEarnings += stakingShareAmount;
```
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:814]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 9]
```diff
@@ -771,17 +783,28 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+          variableToWithdrawnStakingEarningsInShares[msg.sender] += stakingShareAmount.mulDiv(
+            vaultEndingStakesAmount,
+            vaultEndingETHBalance
+          );
+          variableToWithdrawnStakingEarnings[msg.sender] += stakingShareAmount;
```
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:819]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol [hunk 10]
```diff
@@ -771,17 +783,28 @@ contract LidoVault is ILidoVaultInitializer, ILidoVault {
...
+        }
```
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:820]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 11]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
             await lidoVault.connect(addr2).withdraw(SIDE.VARIABLE)
             expect(await lidoVault.vaultEndedWithdrawalsFinalized()).to.equal(false)
 
+            await lidoVault.connect(addr1).finalizeVaultEndedWithdrawals(SIDE.FIXED)
```
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6785]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 12]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
...
             balanceBefore = await ethers.provider.getBalance(addr2)
+            console.log(await lidoVault.withdrawnStakingEarnings() , await lidoVault.vaultEndedStakingEarnings())
+            console.log(await lidoVault.vaultEndingETHBalance() * (await lidoVault.withdrawnStakingEarningsInStakes() -
+            await lidoVault.ongoingProtocolFeeInShares())/ await lidoVault.vaultEndingStakesAmount())
```
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:6789]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 13]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
...
             const receipt = await (
               await lidoVault.connect(addr2).finalizeVaultEndedWithdrawals(SIDE.VARIABLE)
             ).wait()
             const gasFees = calculateGasFees(receipt)
             const balanceAfter = await ethers.provider.getBalance(addr2)
 
             expect(await lidoVault.vaultEndedWithdrawalsFinalized()).to.equal(true)
+            expect(await lidoVault.variableBearerToken(addr2.address)).to.equal(0)
             expect(balanceAfter).to.equal(balanceBefore - gasFees)
```
Comment (sherlock-admin3) [inline lido-fiv/test/1.LidoVault.test.ts:6798]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 14]
```diff
@@ -6782,16 +6782,117 @@ describe('LidoVault', function () {
...
+
+            const balanceAddr3Before = await ethers.provider.getBalance(addr3)
+
+            await expect(
+              lidoVault.connect(addr3).withdraw(SIDE.VARIABLE)
+            ).not.to.be.revertedWith('ETF')
+            await lidoVault.connect(addr3).finalizeVaultOngoingVariableWithdrawals()
+            console.log(balanceAddr3Before -  await ethers.provider.getBalance(addr3))
+
           })
```
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6896]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts [hunk 15]
```diff
@@ -8033,8 +8134,8 @@ describe('LidoVault', function () {
                 ).wait()
                 const gasFees2 = calculateGasFees(receipt2)
                 const balanceAfter2 = await ethers.provider.getBalance(addr2)
-                expect(balanceAfter2 - balanceBefore2).to.equal(
-                  earningsShare2 - gasFees2
+                expect(balanceAfter2 - balanceBefore2).to.equalWithTolerance(
+                  earningsShare2 - gasFees2, 10
```
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:8138]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 1 above)
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:50]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 2 above)
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:128]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 3 above)
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:182]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 4 above)
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:587]: Change related to https://github.com/saffron-finance/lido-fiv/pull/30. 

Fixes issues: [#105](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/105)

File: lido-fiv/contracts/LidoVault.sol (hunk 5 above)
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:799]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 6 above)
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:806]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 7 above)
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:813]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 8 above)
Comment (sherlock-admin4) [inline lido-fiv/contracts/LidoVault.sol:814]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 9 above)
Comment (sherlock-admin2) [inline lido-fiv/contracts/LidoVault.sol:819]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/contracts/LidoVault.sol (hunk 10 above)
Comment (sherlock-admin3) [inline lido-fiv/contracts/LidoVault.sol:820]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 11 above)
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6785]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 12 above)
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:6789]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 13 above)
Comment (sherlock-admin3) [inline lido-fiv/test/1.LidoVault.test.ts:6798]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 14 above)
Comment (sherlock-admin4) [inline lido-fiv/test/1.LidoVault.test.ts:6896]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

File: lido-fiv/test/1.LidoVault.test.ts (hunk 15 above)
Comment (sherlock-admin2) [inline lido-fiv/test/1.LidoVault.test.ts:8138]: Change related to https://github.com/saffron-finance/lido-fiv/pull/29. 

Fixes issues: [#73](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/73), [#90](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/90), [#85](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/85), [#129](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/129), [#87](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/87), [#144](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/144), [#136](https://github.com/sherlock-audit/2024-08-saffron-finance-judging/issues/136)

**Pull request closed without merging**